beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -855,6 +856,44 @@ func (bs *Server) GetBlockRoot(w http.ResponseWriter, r *http.Request) {
 	http2.WriteJson(w, response)
 }
 
+// GetStateFork returns Fork object for state with given 'stateId'.
+func (bs *Server) GetStateFork(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetStateFork")
+	defer span.End()
+	stateId := mux.Vars(r)["state_id"]
+	if stateId == "" {
+		http2.HandleError(w, "state_id is required in URL params", http.StatusBadRequest)
+		return
+	}
+	st, err := bs.Stater.State(ctx, []byte(stateId))
+	if err != nil {
+		http2.HandleError(w, helpers.PrepareStateFetchGRPCError(err).Error(), http.StatusBadRequest)
```

Comment:
- `PrepareStateFetchError` should be used, not the GRPC version
- since we don't know which code to use, it's safer to use `http.StatusInternalServerError`, and that's what we do in other places